signer/tests/integration/transaction_signer.rs

```diff
@@ -359,3 +363,91 @@ async fn handle_pending_deposit_request_address_script_pub_key() {
 
     testing::storage::drop_db(db).await;
 }
+
+/// Test that [`TxSignerEventLoop::assert_valid_stacks_tx_sign_request`]
+/// errors when the signer is not in the signer set.
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn signing_set_validation_check_for_stacks_transactions() {
+    let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
+    let db = testing::storage::new_test_database(db_num, true).await;
+
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+
+    let ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_first_bitcoin_core_client()
+        .with_mocked_emily_client()
+        .with_mocked_stacks_client()
+        .build();
+    let (rpc, faucet) = sbtc::testing::regtest::initialize_blockchain();
+
+    // This confirms a deposit transaction, and has a nice helper function
+    // for storing a real deposit.
+    let setup = TestSweepSetup::new_setup(rpc, faucet, 10000, &mut rng);
+
+    // Let's get the blockchain data into the database.
+    let chain_tip: BitcoinBlockHash = setup.sweep_block_hash.into();
+    backfill_bitcoin_blocks(&db, rpc, &chain_tip).await;
+
+    // This is all normal happy path things that need to happen in order to
+    // pass validation.
+    setup.store_deposit_tx(&db).await;
+    setup.store_sweep_tx(&db).await;
+    setup.store_dkg_shares(&db).await;
+    setup.store_deposit_request(&db).await;
+    setup.store_deposit_decisions(&db).await;
+
+    let (mut req, _) = crate::complete_deposit::make_complete_deposit(&setup);
+
+    req.deployer = ctx.config().signer.deployer;
+    let network = InMemoryNetwork::new();
+    let mut tx_signer = TxSignerEventLoop {
+        network: network.connect(),
+        context: ctx.clone(),
+        context_window: 10000,
+        blocklist_checker: Some(()),
+        wsts_state_machines: HashMap::new(),
+        signer_private_key: setup.aggregated_signer.keypair.secret_key().into(),
+        threshold: 2,
+        rng: rand::rngs::StdRng::seed_from_u64(51),
+    };
+
+    // Let's create a proper sign request.
+    let request = StacksTransactionSignRequest {
+        aggregate_key: setup.aggregated_signer.keypair.public_key().into(),
+        contract_call: signer::stacks::contracts::ContractCall::CompleteDepositV1(req),
+        // The nonce and tx_fee aren't really validated against anything at
+        // the moment.
+        nonce: 1,
+        tx_fee: 100_000,
+        // This can probably be removed, but it's not important to remove
+        // now, we should switch to protobuf messages and remove it then.
```

Comment:
This comment might be fine, but if you haven't already consider if this is worth making a tech-debt type issue for to track, and if so it would be good to link it here with a `TODO`

Reply:
I'm going to fix #517 and remove this.

Reply:
Actually, I'm just going to add a TODO https://github.com/stacks-network/sbtc/pull/694/commits/c3ceea7f8a43bcdd8ef975170e6c5d9aaf965461.